SCons/Tool/dmd.py

```diff
@@ -87,18 +87,20 @@ def generate(env) -> None:
 
     static_obj.add_action('.d', SCons.Defaults.DAction)
     shared_obj.add_action('.d', SCons.Defaults.ShDAction)
-    static_obj.add_emitter('.d', SCons.Defaults.StaticObjectEmitter)
-    shared_obj.add_emitter('.d', SCons.Defaults.SharedObjectEmitter)
+    static_obj.add_emitter('.d', DCommon.DStaticObjectEmitter)
+    shared_obj.add_emitter('.d', DCommon.DSharedObjectEmitter)
 
     env['DC'] = env.Detect(['dmd', 'ldmd2', 'gdmd']) or 'dmd'
-    env['DCOM'] = '$DC $_DINCFLAGS $_DVERFLAGS $_DDEBUGFLAGS $_DFLAGS -c -of$TARGET $SOURCES'
+    env['DCOM'] = '$DC $_DINCFLAGS $_DVERFLAGS $_DDEBUGFLAGS $_DFLAGS $_DINTFDIR -c -of$TARGET $SOURCES'
     env['_DINCFLAGS'] = '${_concat(DINCPREFIX, DPATH, DINCSUFFIX, __env__, RDirs, TARGET, SOURCE)}'
     env['_DVERFLAGS'] = '${_concat(DVERPREFIX, DVERSIONS, DVERSUFFIX, __env__)}'
     env['_DDEBUGFLAGS'] = '${_concat(DDEBUGPREFIX, DDEBUG, DDEBUGSUFFIX, __env__)}'
+    env['_DINTFDIR'] = '${_optWithIxes(DINTFDIRPREFIX, DINTFDIRKEY, DINTFDIRSUFFIX, __env__, Dirs, TARGET, SOURCE)}'
```

Comment:
Can you explain why the normal _concat wouldn't' do here?

Reply:
No not at the moment, I'll test it with concat, thanks.

Reply:
Made it work with concat, although there should only be one DINTFDIR unlike DFLAGS etc.